### ISSUE TITLE
Consistency of user-facing output names (affects API)

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/rail_attic#52]
As part of #9, [I noticed a naming concern that deserves a separate issue](https://github.com/LSSTDESC/RAIL/pull/9#discussion_r583887456). The dictionary keys/column names of the outputs of the creator don't match [those of the estimator](https://github.com/LSSTDESC/RAIL/blob/issue/7/creation-baseline/rail/estimation/estimator.py), which could cause problems for an evaluator seeking to make a direct comparison. I propose the following that would be shared among both: 
- `z_pt` for a Creator's true redshift (which I think is currently taken directly from the input data, i.e. `redshift_true` for the Buzzard demo data) and an Estimator's "best" point estimate (which is currently `zmode` but should be left generic for codes that have a different summarization, since we can always impose our own from the output posteriors)
- `z_posterior_fmt` for the parameterization of the output posteriors, in this case ["interp_pdf"](https://github.com/LSSTDESC/qp/blob/master/qp/interp_pdf.py) for all current Creator and Estimator subclasses
- `z_posterior_metapars` for a Creator's current `pz_grid` and an Estimator's current `zgrid`
- `z_posterior_galpars` for what's currently `pz_pdf` for both the Creator and Estimator superclasses

While `z_posterior_fmt`, `z_posterior_metapars`, and `z_posterior_galpars` would ultimately be replaced with `z_posterior` for the qp.Ensemble defined by its parameterization, catalog-level metaparameters, and per-galaxy parameters (and possibly `z_pt` and the photometry columns as metadata files?) after #33 is resolved, I think enforcing consistency between rail/creation/creator.py and rail/estimation/estimator.py (plus the tests for both) would help for #4.

**Comment by eacharles** [comment]
overtaken by events
